src/tests/mod.rs

```diff
@@ -0,0 +1,9 @@
+mod modules;
```

Comment:
Every test within `src/tests/*` needs to acquire the lock, either concurrently or exclusively, in order to guarantee that the ones which acquire it exclusively are actually executing alone. Currently, there are several tests in these modules that do not acquire the lock.

Reply:
Why would we need to acquire a lock if the test does not interfere with the counter field? I added locks to all tests that create a new object using Object::New() or interfere with counter otherwise.

Reply:
If those tests do not acquire the lock to run concurrently, then the tests that acquire the lock exclusively will be running concurrently with them, which is what is supposed to be avoided by the test lock. In practice, it may not have an effect because you manually vetted those tests, but there's no telling that in the future they won't make a call to `ObjectId::new` or that there are other concurrent conditions to take into account. For those reasons, we need to ensure full usage of the lock across all tests, including the concurrent ones.

Reply:
Thanks, fixed!

Reply:
Looks good, I think just a single test still needs the lock (`src/tests/spec/corpus.rs`).

Reply:
added!